hypothesis-python/tests/conjecture/test_engine.py

```diff
@@ -216,7 +216,7 @@ def f(data):
         data.mark_interesting()
 
     runner = ConjectureRunner(f, settings=settings(max_examples=5000, database=None))
-    with buffer_size_limit(2):
+    with buffer_size_limit(4):
```

Comment:
BUFFER_SIZE is now based on `ir_size`, which includes the additional overhead of the tag bit. Honestly I think we should change this to a custom `choice_size(choice, ir_type, kwargs)` that takes kwargs like min_value into account and does not add the additional tag bit overhead, but that's not critical. (doing this avoids things like `draw_integer(min_size=1 << 1000)` consuming the entire BUFFER_SIZE quota in a single draw).

Reply:
I see what you mean... the buffer handled this by having the `IntegersStrategy` handle the offset from zero; would doing so again solve the problem?

(strings and bytes can also be large, but I don't see a corresponding notion of "offset from zero" for them; there's the possibility of inflation relative to an actually-small alphabet but that seems more manageable...)

Reply:
We could define IntegerStrategy as:
```
def do_draw(data): 
    return self.min_value + data.draw_integer(0, self.max_value - self.min_value)
```

which would solve integers, but this takes information away from the typed choice sequence, which now doesn't know about the min_value of integer choices (it is always zero). This doesn't solve any other strategy which calls `draw_integer`, either.

Strings and bytes have `min_size`, which is similarly problematic! Given this, I'd still prefer `choice_size`